core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -93,41 +106,53 @@ object Broker {
    *               id (int), number of endpoints (int), serialized endpoints
    * @return broker object
    */
-  def readFrom(buffer: ByteBuffer): Broker = {
+  def readFrom(buffer: ByteBuffer, expectRack: Boolean = true): Broker = {
     val id = buffer.getInt
     val numEndpoints = buffer.getInt
 
     val endpoints = List.range(0, numEndpoints).map(i => EndPoint.readFrom(buffer))
             .map(ep => ep.protocolType -> ep).toMap
-    new Broker(id, endpoints)
+    if (expectRack) {
+      val rack = readShortString(buffer)
+      new Broker(id, endpoints, Option(rack))
+    } else {
+      new Broker(id, endpoints, Option.empty)
+    }
   }
 }
 
-case class Broker(id: Int, endPoints: Map[SecurityProtocol, EndPoint]) {
+case class Broker(id: Int, endPoints: Map[SecurityProtocol, EndPoint], rack: Option[String]) {
```

Comment:
The KIP says:

`case class Broker(id: Int, endPoints: Map[SecurityProtocol, EndPoint], rack: Option[String] = None)`

I prefer how you have it here, but we should update the KIP.